packages/toasts/src/ToastsProvider.jsx

```diff
@@ -0,0 +1,57 @@
+import React, {
+  createContext, useState, useMemo, useCallback,
+} from 'react';
+import PropTypes from 'prop-types';
+
+export const ToastsContext = createContext();
+const ToastsProvider = ({ children }) => {
+  const [toasts, setToasts] = useState([]);
+  function getRandom(min, max) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+  /**
+   * message parameter can either be a string or React node component - Displays text or component in the toasts.
+   * delay parameter must be a number - Optional. The time in milliseconds to show the toast.
+   * action parameter must be an object with the following properties:
+   * label: string - Required. The label for the action button.
+   * href: string - **Optional**. The href for the action button.
+   * onClick: function - **Optional**. The function to be called when the action button is clicked.
+   * ** action must contain either href or onClick property **.
+   * The functionality mimics the functionality of the '@edx/paragon' Toast component.
+   * See {@link https://paragon-openedx.netlify.app/components/toast/#home}
+   */
+  const addToast = useCallback((message, actionObj = null, delayLength = 5000) => {
+    setToasts(prevToasts => [
+      ...prevToasts,
+      {
+        id: getRandom(0, 1000),
```

Comment:
[Question]: Why are we using a random id here instead of the length of toasts? The id here is not guaranteed to be unique.